builtin/providers/aws/resource_aws_network_acl_rule.go

```diff
@@ -63,12 +63,12 @@ func resourceAwsNetworkAclRule() *schema.Resource {
 				ForceNew: true,
 			},
 			"icmp_type": &schema.Schema{
-				Type:     schema.TypeInt,
+				Type:     schema.TypeString,
 				Optional: true,
 				ForceNew: true,
```

Comment:
Let's also add a `ValidateFunc` to these fields that attempts the `Atoi` and returns an error when parsing fails. This will let users get much faster feedback if they typo something that is not an integer.